core/reactor_common.c

```diff
@@ -549,6 +549,14 @@ void _lf_initialize_timers(environment_t* env) {
             _lf_initialize_timer(env, env->timer_triggers[i]);
         }
     }
+    
+    // Create an extra event and put it on the recycle queue. 
+    // An LF program with n timers require n+1 events and the last event 
+    // needs to be initialized to the event recycle queue.
```

Comment:
Unclear what this last clause means.  Perhaps just state in the comment that the reason is avoid runtime memory allocation for timer-driven programs?

Reply:
Thanks, updated